### DISCUSSION TITLE
ctx meta user issues

### DISCUSSION THREAD

**Discussion by mituBarua** [moleculerjs/moleculer#1108] [General]
How can i access ctx.meta.user ?

**Comment by intech** [comment]
@mituBarua describe how tired you are and get meta so we can help you.
An example code would be ideal for reproducing your problem.

**Comment by ccampanale** [comment]
@mituBarua 

Pretty sure that `ctx.meta.user` is only ever set if something in your app sets it OR if you are using `moleculer-web` and based your app on one of the provided examples. Basically, `ctx.meta.user` may be set ([as seen in this example](https://github.com/moleculerjs/moleculer-web/blob/ce6881069e12819ab1bf890c159b3493c9f908ed/examples/full/index.js#L370)) as part of a call to the API GW's `authorize()` service method since [authorization is enabled for the route](https://github.com/moleculerjs/moleculer-web/blob/ce6881069e12819ab1bf890c159b3493c9f908ed/examples/full/index.js#L134).

Assuming this is actually being set you should be able to access the metadata property on the context object provided either actions handlers and non-legacy event handlers:

[This example](https://github.com/moleculerjs/moleculer-web/blob/ce6881069e12819ab1bf890c159b3493c9f908ed/test/services/test.service.js#L66-L74) shows an action handler using `ctx.meta.user` to retrieve the user's `username`.

**Comment by mituBarua** [comment]
@ccampanale  @intech  whenever i tried to https://github.com/moleculerjs/moleculer-examples/blob/master/conduit/services/users.service.js update this user list or update any service of this repo it is giving thorugh error of meta.user._id is undefined.

**Comment by mituBarua** [comment]
![sn1](https://user-images.githubusercontent.com/64905702/181426888-66498a5a-4be4-4ed6-ac27-844f2214484b.PNG)
![sn2](https://user-images.githubusercontent.com/64905702/181426892-75fcb6ee-50e4-4f20-a17d-79559430f511.PNG)

**Comment by intech** [comment]
These methods require authorization:

https://github.com/moleculerjs/moleculer-examples/blob/master/conduit/services/api.service.js#L72-L99
